### PR TITLE
Banner

### PR DESCRIPTION
**Pull request by kokoras88**
I am sending the banner for PR. Have a look and in the future we change what we want to change

### CODE REVIEW

**Comment by wJoenn** [comment]
Can you share a screenshot of what you've done please ?
You also have conflicts that need to be resolved

**Comment by kokoras88** [comment]
Screenshot from 2023-05-05 16-38-42

**Comment by kokoras88** [comment]
Screenshot from 2023-05-05 16-38-42

File: app/views/shared/_banner.html.erb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<div class="banner"">
+    <nav>
```
Comment (haqcked) [inline app/views/shared/_banner.html.erb:2]: Pls correct me if im wrong, but I think the Navbar should have its own partial which will be rendered in the application.html.erb

Comment (wJoenn) [reply app/views/shared/_banner.html.erb:2]: Not necessarily in the layout itself because there might be some pages where we don't want the navbar.
The admin interface wouldn't need it for example.

But I do agree moving the navbar in a reusable component would be better

Comment (haqcked) [reply app/views/shared/_banner.html.erb:2]: you mean to only render it in the homepage right? not in the pages where it wont be needed.

Comment (wJoenn) [reply app/views/shared/_banner.html.erb:2]: Yes if Alex makes a partial out of the navbar then we can render it only in the pages we want and skip the others 🙂

Comment (kokoras88) [reply app/views/shared/_banner.html.erb:2]: Hello gentlemen! I totally agree that the navbar should be a partial. I just got home from work so i wont work on it today but tomorrow i have office day so i can deal with it ! Have a nice day!

**Comment by kokoras88** [comment]
Guys i did some changes in the navbar. Now it's shared so we can use it with an if wherever we want. In this example is shared inside the banner. Please review the code and let me know! Cheers

File: app/frontend/stylesheets/components/_navbar.scss [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+ul {
+  margin: auto;
+  display: flex;
+  justify-content: space-around;
+  max-width: 400px;
+}
+
+ul li {
```
Comment (StephanK83) [inline app/frontend/stylesheets/components/_navbar.scss:20]: In general I'm a friend of using unique named div class containers if needed and adressing them in the scss, to modify their design. If using ul li p h1 and so on in a navbar.scss it's possible, that it will affect ul li p h1 outside of the _navbar.html.erb. 

Just a proposal.

File: app/frontend/stylesheets/config/_setup.scss [hunk 3]
```diff
@@ -22,6 +28,13 @@ body {
   min-height: 100vh;
 }
 
+
+h1, h2, h3 {
+h1, h2 {
```
Comment (StephanK83) [inline app/frontend/stylesheets/config/_setup.scss:33]: This does not make sense. h1 and h2 are not nested inside h1 and h2 in any html.erb-document I guess :)

File: app/views/pages/home.html.erb [hunk 4]
```diff
@@ -1,3 +1,13 @@
+
+
+<%= render "shared/banner"%>
```
Comment (StephanK83) [inline app/views/pages/home.html.erb:3]: I propose to put everything what is displayed on the page in a div class container.

File: app/views/shared/_banner.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<div class="banner"">
+<%= render "shared/navbar" %>
```
Comment (StephanK83) [inline app/views/shared/_banner.html.erb:2]: Indentation

File: app/views/shared/_banner.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+
+
+    <div class="text">
+        <h1>Your Body, Your Practice</h1>
+        <p>DISCOVER YOURSELF, DISCOVER YOGA</p>
+        <button class="btn btn-gradient">Start now</button>
+    </div>
+
+</div>
+dev
```
Comment (StephanK83) [inline app/views/shared/_banner.html.erb:12]: ?

File: app/views/shared/_navbar.html.erb [hunk 7]
```diff
@@ -0,0 +1,14 @@
+    <div class="navbar">
```
Comment (StephanK83) [inline app/views/shared/_navbar.html.erb:1]: Identation

**Review by StephanK83 — COMMENTED** [review]
See my comments in the code.

File: app/frontend/stylesheets/components/_navbar.scss (hunk 2 above)
Comment (wJoenn) [reply app/frontend/stylesheets/components/_navbar.scss:20]: I agree with Stephan, this is not safe
But instead of giving them all a class, just embed everything inside the .navbar selector
That way it ensures that every propreties you apply to these element only apply if they are inside the navbar

.navbar {
   ...
   
   .logo {
   }
   
   ul {
     li {
       &:not(:first-child)::before {
       }
       
       a {
       }
     }
   }
 }

File: app/frontend/stylesheets/config/_setup.scss (hunk 3 above)
Comment (wJoenn) [reply app/frontend/stylesheets/config/_setup.scss:33]: And you already have the same thing at line 38->40 too

File: app/views/pages/home.html.erb (hunk 4 above)
Comment (wJoenn) [reply app/views/pages/home.html.erb:3]: Not everything otherwise the banner background-image wouldn't  be able to take the whole width.

Comment (StephanK83) [reply app/views/pages/home.html.erb:3]: No, this is not true. Imagine having a parent div class home-container, next there is a div class banner-container with a background image inside. The background-image will take the whole width of it's parent container by default. Only when using margins or paddings for any of these, it will not take the whole width. 

I would be surprised if I was wrong. But I'm open for surprises. :)

File: app/frontend/stylesheets/components/_navbar.scss (hunk 2 above)
Comment (StephanK83) [reply app/frontend/stylesheets/components/_navbar.scss:20]: Yes, this is nice. Looks good. If you do it like this, and you have nothing left, which is not nested in the .navbar you could move the whole content theoretically to application.scss. :)

File: app/views/pages/home.html.erb (hunk 4 above)
Comment (wJoenn) [reply app/views/pages/home.html.erb:3]: Yeah it would take the full width of its parent, the container, but the container doesn't have a width of the whole screen.

The idea of a container is that it contains the code inside a block that is smaller than the screen to focus the content of the page more toward the center with a margin auto
So if you were to have the background image inside the container you would have something like this
image

What we want instead is to have a div.banner that takes the full width of the body and inside that div.banner we add a container to center the content
image


That means that in the code it would look something like
```
body
  div.banner
    div.container
      ...
      ...
    /div
  /div
  
  div.container
    main
  /div
/body
```

File: app/frontend/stylesheets/config/_setup.scss (hunk 3 above)
Comment (kokoras88) [reply app/frontend/stylesheets/config/_setup.scss:33]: I didn't write these lines of code. I know that they dont make sense but i dont know how the got there

Comment (wJoenn) [reply app/frontend/stylesheets/config/_setup.scss:33]: Probably when merging master into your branch

File: app/frontend/stylesheets/components/_navbar.scss (hunk 2 above)
Comment (kokoras88) [reply app/frontend/stylesheets/components/_navbar.scss:20]: Yes i get your point guys. I just nested everything inside the navbar div and  now they don't affect anything else out of the navbar. Tested already!

File: app/frontend/stylesheets/config/_setup.scss (hunk 3 above)
Comment (kokoras88) [reply app/frontend/stylesheets/config/_setup.scss:33]: I just deleted line 32 to 36. 38 to 40 is making the job i believe

File: app/views/pages/home.html.erb (hunk 4 above)
Comment (StephanK83) [reply app/views/pages/home.html.erb:3]: So imagine:

`<body>
  <div class="container-page-content">
    <div class="container-banner">
    </div>
  </div>
</body>`

or

`<body>
  <div class="container-banner">
  </div>
  <div class="container-page-content">
  </div>
</body>`

container-banner would have the full width of container-page-content in the first example, which has the full width of body, unless no different styles appplied.

in second example container-banner and container-page-content would have the same width as body if not styled different.

The idea of that divs is, to structure content, to control layout, style and maybe js-behaviour on point and of course to increase readability. That's why whenever I was doing something on that pages, I wrapped everything in divs.

The graphics you used do not show the real behaviour (if not styled with css like this), they are just to show the structure and relation between body and containers.

In the second example the banner is wrapped in the body. If you don't size the body width with css, it has the whole screen width. same if you wrap a container (which contains the banner) in the body and don't size it with css it has the whole width by default.

That is my understanding. What else should be the default size of a container wrapped in the body if no style is applied?

Comment (kokoras88) [reply app/views/pages/home.html.erb:3]: I didn't change anything as it looks good so far. If we dont like smth in the future we can try to do it

File: app/views/shared/_banner.html.erb (hunk 6 above)
Comment (kokoras88) [reply app/views/shared/_banner.html.erb:12]: I sent it to Valhala

**Comment by kokoras88** [comment]
I reviewed your comments and fixed the issues. If you need to change smth else let me know

File: app/views/pages/home.html.erb (hunk 4 above)
Comment (StephanK83) [reply app/views/pages/home.html.erb:3]: Sorry, tried to make my position understandable, but it did not show the code, that's why I deleted my text, copied it before, but it got lost, now I need to go. :D :D :D 

But as I've seen, you read it. 

To follow a structure, put everything in divs, that was the idea. But we can leave it like this and finally merge. :)

"Everything that should not be wider than 1200px or 80% should be contained in a .container like this one, the membership cards for example are inside of it because we don't want the cards to take the width of the whole screen"

Okay, this is an interesting logic. Sounds good.

**Review by k-kenney — APPROVED** [review]
Well done. The changes look nice and clean. I'm approving this.

File: app/frontend/stylesheets/config/_colors.scss [hunk 8]
```diff
@@ -0,0 +1,10 @@
+// Define variables for your color scheme
+
+// For example:
+$red: #FD1015;
+$blue: #0D6EFD;
+$yellow: #FFC65A;
+$orange: #E67E22;
+$green: #1EDD88;
+$gray: #0E0000;
+$light-gray: #F4F4F4;
```
Comment (wJoenn) [inline app/frontend/stylesheets/config/_colors.scss:10]: Colours are already being defined inside the config/_variables.scss file.
Can you move any colour you need inside of there and delete this file please ?

File: app/frontend/stylesheets/config/_fonts.scss [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+// To use a font file (.woff) uncomment following lines
+// @font-face {
+//   font-family: "Font Name";
+//   src: font-url('FontFile.eot');
+//   src: font-url('FontFile.eot?#iefix') format('embedded-opentype'),
+//        font-url('FontFile.woff') format('woff'),
+//        font-url('FontFile.ttf') format('truetype')
+// }
+// $my-font: "Font Name";
```
Comment (wJoenn) [inline app/frontend/stylesheets/config/_fonts.scss:18]: Fonts have already been imported inside the config/_variables.scss and the $body-font and $headers-font variables have already been defined so you can delete this file.

I think the fonts are not the same as the one you chose but they're the ones approved by Chloe, if we realize they don't fit as well as we'd like we can change them back later by changing the google import inside the config/_variables.scss file

File: app/frontend/stylesheets/application.scss [hunk 10]
```diff
@@ -1,5 +1,11 @@
 // Config files
+
+@import "config/fonts";
+@import "config/colors";
+
 @import "config/variables";
+
+
 @import "config/setup";
```
Comment (wJoenn) [inline app/frontend/stylesheets/application.scss:9]: You can remove everything you added here after deleting the fonts and colors file like I asked.
There should only be the variables and setup import in the `// Config File` section of this file after that

File: app/frontend/stylesheets/components/_button.scss [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+  border: 1px solid #30973e;
+  font-size: 20px;
+  box-shadow: 0px 8px 15px rgba(0, 0, 0, 0.2);
+}
+
+.btn-gradient:hover {
+  background: linear-gradient(#587553, #0d9651);
+  color: white;
+  cursor: pointer;
+}
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_button.scss:17]: There already is a components/_buttons.scss file so please delete this one.
For consistency it would be better if you used the existing button classes such as .btn and .btn-green to avoid having buttons with different style all over the place.

File: app/frontend/stylesheets/components/_banner.scss [hunk 12]
```diff
@@ -0,0 +1,37 @@
+.banner {
+  height: 90vh;
+  background-image: url("../../images/banner.jpg");
+  background-size: cover;
+  background-position: center -260px;
+}
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_banner.scss:6]: This banner should be 100vh height

File: app/frontend/stylesheets/components/_banner.scss [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+//----------------------nav-----------------------//
+
+
+
+
+
+//---------------------------------text-----------------------------//
+
+
+
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_banner.scss:20]: Can you remove this empty space please ?

File: app/frontend/stylesheets/config/_setup.scss [hunk 14]
```diff
@@ -1,3 +1,9 @@
+
+// body {
+//   background-color: #181a1b;
+//   color: lightgrey;
+// }
+
```
Comment (wJoenn) [inline app/frontend/stylesheets/config/_setup.scss:6]: Can you remove this comment please ?

File: app/frontend/stylesheets/config/_setup.scss [hunk 15]
```diff
@@ -22,6 +28,8 @@ body {
   min-height: 100vh;
 }
 
+
+
```
Comment (wJoenn) [inline app/frontend/stylesheets/config/_setup.scss:32]: Can you remove these empty lines please ?

File: app/views/shared/_banner.html.erb [hunk 16]
```diff
@@ -0,0 +1,8 @@
+<div class="banner"">
+    <%= render "shared/navbar" %>
+    <div class="text">
+        <h1>Your Body, Your Practice</h1>
+        <p>DISCOVER YOURSELF, DISCOVER YOGA</p>
+        <button class="btn btn-gradient">Start now</button>
+    </div>
+</div>
```
Comment (wJoenn) [inline app/views/shared/_banner.html.erb:8]: The indentation should be 2 spaces, not 4

also please add an empty space between the navbar and the div.text to space this html a bit. It'll make it easier to read.
You also need to wrap the navbar and the div.text inside a div.container

Also this partial will only every be used inside the home page so please move it inside the views/pages/partials folder

File: app/views/shared/_navbar.html.erb [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+
+        <div class="links">
+          <ul>
+            <li><a href="#">About</a></li>
+            <li><a href="#">Sessions</a></li>
+            <li><a href="#">Contact</a></li>
+            <li><a href="#">Videos</a></li>
+          </ul>
+        </div>
+    </div>
```
Comment (wJoenn) [inline app/views/shared/_navbar.html.erb:14]: Again the indentation should be 2 spaces, not 4
Also because your ul is a list of links you should use a <nav> tag instead. And inside the nav you can just use <a> tags directly wihout <li>

File: app/views/pages/home.html.erb (hunk 4 above)
Comment (wJoenn) [inline app/views/pages/home.html.erb:3]: Please remove the empty spaces at the top and add a space between "shared/banner" and %>

File: app/views/pages/home.html.erb [hunk 18]
```diff
@@ -1,3 +1,13 @@
...
+
+<h1 data-controller="hello">Pages#home</h1>
+<p>Find me in app/views/pages/home.html.erb</p>
+<%= render "pages/partials/membership" %>
+
+
+
```
Comment (wJoenn) [inline app/views/pages/home.html.erb:10]: This is a duplicate of the code at line 11 to 20. Please remove it

File: app/views/pages/home.html.erb [hunk 19]
```diff
@@ -8,3 +18,4 @@
     </section>
   </div>
 </div>
+</div>
```
Comment (wJoenn) [inline app/views/pages/home.html.erb:21]: Please delete this line

File: app/frontend/stylesheets/components/_navbar.scss [hunk 20]
```diff
@@ -0,0 +1,42 @@
+
+.navbar {
+  margin-left: 300px;
+  display: flex;
+  // justify-content: space-around;
+  align-items: center;
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_navbar.scss:6]: Remove the comment please.
And don't use a fixed margin because this is not reactive at all. Use justify-content: center instead when you want to center something on the x axis

File: app/frontend/stylesheets/components/_navbar.scss [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+      }
+
+      ul li a {
+        margin-left: 20px;
+        text-decoration: none;
+        color: #fff;
+        position: relative;
+        font-size: 20px;
+        font-size: 18px;
+      }
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_navbar.scss:41]: The indentation isn't right.

Also please use SCSS' nesting like so
```
ul {
  ...
  
  li {
    ...
    
    &:not(:first-child)::before {
      ...
    }
    
    a {
      ...
    }
  }
}
```

Instead of using margin-right on the .logo use gap on the .navbar

- You can remove the max-width and the margin auto from the ul
- The list-style of the li is already none from the config/_setup.scss file
- give the padding of the li to the ul instead and give the ul a gap to create spacing between the li
- Do not use bottom on the li
- ::before will be difficult to be responsive. Just add <li>*</li> between the links you already have in the html that'll be easier to manage
- don't use margin or position on the a tags
- text-decoration is already set in the config/_setup so you can remove it
- you have 2 font-sizes in the a tag, remove one

File: app/frontend/stylesheets/components/_banner.scss [hunk 22]
```diff
@@ -0,0 +1,37 @@
...
+    color: #fff;
+  }
+
+  p{
+    font-size: 15px;
+    font-weight: bold;
+    color: #fff;
+    margin-bottom: 40px;
+  }
+}
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_banner.scss:37]: - Move this inside the .banner css selector
- Don't use fixed value margin here, use flex instead to position this
- You don't need to set the font-family to be $header-font, this is already done in the config/_setup.scss file
- Don't use margins to create distance between elements. Turn the div.text into a display flex instead and use gap
- Move the color to the .text selector, it'll apply to all children

**Review by wJoenn — CHANGES_REQUESTED** [review]

File: app/frontend/images/banner.jpg
[the remark this answers is not in the document]
Comment (kokoras88) [reply app/frontend/images/banner.jpg:1]: I guess you mean the banner.jpg

File: app/frontend/stylesheets/application.scss (hunk 10 above)
Comment (kokoras88) [reply app/frontend/stylesheets/application.scss:9]: Ok

File: app/views/pages/home.html.erb (hunk 18 above)
Comment (kokoras88) [reply app/views/pages/home.html.erb:10]: I delleted line 8

File: app/views/pages/home.html.erb (hunk 19 above)
Comment (kokoras88) [reply app/views/pages/home.html.erb:21]: Done

File: app/frontend/stylesheets/components/_navbar.scss (hunk 20 above)
Comment (kokoras88) [reply app/frontend/stylesheets/components/_navbar.scss:6]: Sure! My bad

File: app/frontend/images/banner.jpg
[the remark this answers is not in the document]
Comment (wJoenn) [reply app/frontend/images/banner.jpg:1]: Right 😅

File: app/frontend/stylesheets/config/_colors.scss (hunk 8 above)
Comment (wJoenn) [reply app/frontend/stylesheets/config/_colors.scss:10]: Yes but not all of them. Only the one you actually use. There's no point in having 20 variables we never use in the app

**Comment by kokoras88** [comment]
I have finished some changes in the banners code. Please review  and let me know!

File: app/frontend/stylesheets/components/_banner.scss [hunk 23]
```diff
@@ -0,0 +1,29 @@
+.banner {
+  height: 100vh;
+  background-image: url("../../images/photos/banner.jpg");
+  background-size: cover;
+  background-position: center -260px;
+
+  .text {
+    display: grid;
+    grid-auto-flow: row;
+    grid-row-gap: 60px;
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_banner.scss:10]: Change this to a display flex please

File: app/frontend/stylesheets/components/_banner.scss [hunk 24]
```diff
@@ -0,0 +1,29 @@
...
+      font-size: 40px;
+      padding: 5px;
+      color: #fff;
+    }
+
+    p{
+      font-size: 15px;
+      font-weight: bold;
+      color: #fff;
+    }
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_banner.scss:23]: move the color to the .text please

File: app/frontend/stylesheets/components/_banner.scss [hunk 25]
```diff
@@ -0,0 +1,29 @@
...
+
+    .btn {
+      max-width: 100px;
+    }
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_banner.scss:27]: change this to width: max-content

File: app/frontend/stylesheets/components/_banner.scss [hunk 26]
```diff
@@ -0,0 +1,29 @@
...
+  }
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_banner.scss:28]: This text's position doesn't look right on my 1440p screen
Can you make it responsive to always be in the middle left please ?
image

File: app/frontend/stylesheets/components/_index.scss [hunk 27]
```diff
@@ -1 +1,4 @@
+
+@import "banner";
 @import "buttons";
+@import "navbar";
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_index.scss:4]: remove the empty line a line 1

File: app/frontend/stylesheets/components/_navbar.scss [hunk 28]
```diff
@@ -0,0 +1,33 @@
...
+        display: flex;
+        align-items: center;
+      }
+
+      span {
+        margin-top: 12px;
+        color: #fff;
+        font-size: 20px;
+      }
+}
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_navbar.scss:33]: remove the empty line at line 1 and fix the indentation

File: app/frontend/stylesheets/config/_setup.scss [hunk 29]
```diff
@@ -1,3 +1,4 @@
...
```
Comment (wJoenn) [inline app/frontend/stylesheets/config/_setup.scss:1]: Can you remove the empty lines at line 1, 26 and 27 ?

File: app/frontend/stylesheets/config/_variables.scss [hunk 30]
```diff
@@ -3,8 +3,10 @@
 // Borders & Box Shadows
 
 // Colours
-$blue: #7f8ea8;
-$green: #85b6aa;
+
+$blue: #0D6EFD;
+$green: #1EDD88;
+
```
Comment (wJoenn) [inline app/frontend/stylesheets/config/_variables.scss:9]: Why did you change the existing colours ?

File: app/views/pages/home.html.erb [hunk 31]
```diff
@@ -1,3 +1,12 @@
+
+
+<%= render "pages/partials/banner"%>
+
+<h1 data-controller="hello">Pages#home</h1>
+<p>Find me in app/views/pages/home.html.erb</p>
+
+
+
```
Comment (wJoenn) [inline app/views/pages/home.html.erb:9]: the banner should be rendered inside the #home element
And you can remove the h1 and the p

File: app/views/pages/home.html.erb [hunk 32]
```diff
@@ -8,3 +17,4 @@
     </section>
   </div>
 </div>
+</div>
```
Comment (wJoenn) [inline app/views/pages/home.html.erb:20]: Delete this line 20 please, the `</div>` is not connected to anything

File: app/views/pages/partials/_banner.html.erb [hunk 33]
```diff
@@ -0,0 +1,12 @@
+<div class="banner"">
+    <div class="container">
+      <%= render "shared/navbar" %>
+
+      <div class="text">
+        <h1>Your Body, Your Practice</h1>
+        <p>DISCOVER YOURSELF, DISCOVER YOGA</p>
+        <button class="btn btn-gradient">Start now</button>
```
Comment (wJoenn) [inline app/views/pages/partials/_banner.html.erb:8]: You don't need a .btn class on a button

File: app/views/shared/_navbar.html.erb [hunk 34]
```diff
@@ -0,0 +1,14 @@
...
+
+        <div class="links">
+          <ul>
+            <li><a href="#">About</a></li> <span>*</span>
+            <li><a href="#">Sessions</a></li> <span>*</span>
+            <li><a href="#">Contact</a></li> <span>*</span>
+            <li><a href="#">Videos</a></li>
+          </ul>
+        </div>
+    </div>
```
Comment (wJoenn) [inline app/views/shared/_navbar.html.erb:14]: Indentation still isn't correct

**Review by wJoenn — CHANGES_REQUESTED** [review]

**Review by wJoenn — CHANGES_REQUESTED** [review]

File: app/frontend/stylesheets/components/_navbar.scss [hunk 35]
```diff
@@ -0,0 +1,32 @@
...
+      display: flex;
+      align-items: center;
+    }
+
+    span {
+      margin-top: 12px;
+      color: #fff;
+      font-size: 20px;
+    }
+}
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_navbar.scss:32]: indentation still isn't correct

File: app/frontend/stylesheets/config/_setup.scss [hunk 36]
```diff
@@ -22,6 +22,8 @@ body {
   min-height: 100vh;
 }
 
+
+
 h1, h2, h3 {
   font-family: $headers-font;
 }
```
Comment (wJoenn) [inline app/frontend/stylesheets/config/_setup.scss:29]: There should only be one empty line between selectors. Remove line 25 and 26 please

File: app/frontend/stylesheets/config/_variables.scss [hunk 37]
```diff
@@ -3,9 +3,11 @@
 // Borders & Box Shadows
 
 // Colours
+
 $blue: #7f8ea8;
 $green: #85b6aa;
 
+
 // Spacing
 $padding: 0.5rem 1rem;
```
Comment (wJoenn) [inline app/frontend/stylesheets/config/_variables.scss:13]: There should not be an empty line after // Colours and there should only be 1 empty line between the colours and the // Spacing.
Remove linbe 6 and 10 please

File: app/views/pages/home.html.erb [hunk 38]
```diff
@@ -1,8 +1,10 @@
...
+  <%= render "pages/partials/banner"%>
   <div class="container">
-    <%= render "pages/partials/membership" %>
+  <%= render "pages/partials/membership" %>
 
     <section id="planning">
+
       <h2>Planning</h2>
       <%= render "pages/partials/calendar" %>
     </section>
```
Comment (wJoenn) [inline app/views/pages/home.html.erb:10]: The indentation isn't correct.
Can you also rmeove the empty line after the section.planning and add an empty line before the div.container please ?

File: app/views/pages/partials/_banner.html.erb [hunk 39]
```diff
@@ -0,0 +1,11 @@
...
+  <div class="container">
+  <%= render "shared/navbar" %>
+
+  <div class="text">
+    <h1>Your Body, Your Practice</h1>
+      <p>DISCOVER YOURSELF, DISCOVER YOGA</p>
+      <button>Start now</button>
+    </div>
+  </div>
+</div>
```
Comment (wJoenn) [inline app/views/pages/partials/_banner.html.erb:11]: The indentation isn't correct

File: app/views/shared/_navbar.html.erb [hunk 40]
```diff
@@ -0,0 +1,14 @@
+<div class="navbar">
+  <div class="logo">
+  <%= vite_image_tag "images/logos/logo_text_white.png", width: "180px" %>
+  </div>
```
Comment (wJoenn) [inline app/views/shared/_navbar.html.erb:4]: The image need to be indented

**Review by wJoenn — CHANGES_REQUESTED** [review]

File: app/views/shared/_navbar.html.erb [hunk 41]
```diff
@@ -0,0 +1,14 @@
...
+
+  <div class="links">
+    <ul>
+      <li><a href="#">About</a></li> <span>*</span>
+      <li><a href="#">Sessions</a></li> <span>*</span>
+      <li><a href="#">Contact</a></li> <span>*</span>
+      <li><a href="#">Videos</a></li>
+    </ul>
+  </div>
```
Comment (wJoenn) [inline app/views/shared/_navbar.html.erb:13]: You forgot to change this into a <nav> instead of a <ul>

File: app/views/pages/partials/_banner.html.erb [hunk 42]
```diff
@@ -0,0 +1,11 @@
...
+  <div class="container">
+    <%= render "shared/navbar" %>
+
+    <div class="text">
+      <h1>Your Body, Your Practice</h1>
+        <p>DISCOVER YOURSELF, DISCOVER YOGA</p>
+        <button>Start now</button>
+    </div>
+  </div>
+</div>
```
Comment (wJoenn) [inline app/views/pages/partials/_banner.html.erb:11]: The indentation sill isn't correct

File: app/frontend/stylesheets/config/_variables.scss [hunk 43]
```diff
@@ -1,14 +1,11 @@
...
+
 $blue: #7f8ea8;
-$green: #85b6aa;
 
 // Spacing
 $padding: 0.5rem 1rem;
-
 // Fonts'Comfortaa', cursive;
 $body-font: 'Comfortaa', cursive;
 $headers-font: 'Poppins', sans-serif;
```
Comment (wJoenn) [inline app/frontend/stylesheets/config/_variables.scss:11]: - [ ] There should be an empty space after the import
- [ ] There should be an empty space after // Border
- [ ] The should not be an empty space after // Colours
- [ ] You should not have deleted the green
- [ ] There should be an empty space after $padding

**Review by wJoenn — CHANGES_REQUESTED** [review]

File: app/frontend/stylesheets/components/_navbar.scss [hunk 44]
```diff
@@ -0,0 +1,33 @@
...
+
+
+  nav li a {
+    margin-left: 20px;
+    color: #fff;
+    position: relative;
+    font-size: 20px;
+    display: flex;
+    align-items: center;
+    }
```
Comment (wJoenn) [inline app/frontend/stylesheets/components/_navbar.scss:26]: Why are there 2 lines between these 2 selectors ?
And the indentation of the closing bracket of your `nav li a` selector isn't correct

File: app/views/shared/_navbar.html.erb [hunk 45]
```diff
@@ -0,0 +1,14 @@
...
+    <%= vite_image_tag "images/logos/logo_text_white.png", width: "180px" %>
+  </div>
+
+  <div class="links">
+    <nav>
+      <li><a href="#">About</a></li> <span>*</span>
+      <li><a href="#">Sessions</a></li> <span>*</span>
+      <li><a href="#">Contact</a></li> <span>*</span>
+      <li><a href="#">Videos</a></li>
+    </nav>
```
Comment (wJoenn) [inline app/views/shared/_navbar.html.erb:12]: Don't use li tags inside a `<nav>`
Also please make each span take their own line

**Review by wJoenn — APPROVED** [review]
